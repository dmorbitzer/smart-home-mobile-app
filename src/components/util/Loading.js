import { Text, Box } from "@react-native-material/core";
import { ActivityIndicator } from "react-native";

export default function Loading() {
  return (
    <Box style={{ alignSelf: "center", marginTop: 40 }}>
      <ActivityIndicator size="large" />
      <Box m={10}>
        <Text variant="h6">Loading</Text>
      </Box>
    </Box>
  );
}
