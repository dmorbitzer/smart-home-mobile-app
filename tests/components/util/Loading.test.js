import React from "react";
import renderer from "react-test-renderer";

import Loading from "../../../src/components/util/Loading";

test("renders Loading correctly", () => {
  const tree = renderer.create(<Loading />).toJSON();
  expect(tree).toMatchSnapshot();
});
