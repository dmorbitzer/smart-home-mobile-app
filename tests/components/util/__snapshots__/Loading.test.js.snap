// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Loading correctly 1`] = `
<View
  style={
    [
      {
        "backgroundColor": undefined,
        "borderBottomColor": undefined,
        "borderBottomEndRadius": undefined,
        "borderBottomLeftRadius": undefined,
        "borderBottomRightRadius": undefined,
        "borderBottomStartRadius": undefined,
        "borderBottomWidth": undefined,
        "borderColor": undefined,
        "borderEndColor": undefined,
        "borderEndWidth": undefined,
        "borderLeftColor": undefined,
        "borderLeftWidth": undefined,
        "borderRadius": undefined,
        "borderRightColor": undefined,
        "borderRightWidth": undefined,
        "borderStartColor": undefined,
        "borderStartWidth": undefined,
        "borderStyle": undefined,
        "borderTopColor": undefined,
        "borderTopEndRadius": undefined,
        "borderTopLeftRadius": undefined,
        "borderTopRightRadius": undefined,
        "borderTopStartRadius": undefined,
        "borderTopWidth": undefined,
        "borderWidth": undefined,
        "bottom": undefined,
        "end": undefined,
        "height": undefined,
        "left": undefined,
        "margin": undefined,
        "marginBottom": undefined,
        "marginEnd": undefined,
        "marginHorizontal": undefined,
        "marginLeft": undefined,
        "marginRight": undefined,
        "marginStart": undefined,
        "marginTop": undefined,
        "marginVertical": undefined,
        "maxHeight": undefined,
        "maxWidth": undefined,
        "minHeight": undefined,
        "minWidth": undefined,
        "opacity": undefined,
        "overflow": undefined,
        "padding": undefined,
        "paddingBottom": undefined,
        "paddingEnd": undefined,
        "paddingHorizontal": undefined,
        "paddingLeft": undefined,
        "paddingRight": undefined,
        "paddingStart": undefined,
        "paddingTop": undefined,
        "paddingVertical": undefined,
        "position": undefined,
        "right": undefined,
        "start": undefined,
        "top": undefined,
        "width": undefined,
        "zIndex": undefined,
      },
      {
        "alignSelf": "center",
        "marginTop": 40,
      },
    ]
  }
>
  <ActivityIndicator
    size="large"
  />
  <View
    style={
      [
        {
          "backgroundColor": undefined,
          "borderBottomColor": undefined,
          "borderBottomEndRadius": undefined,
          "borderBottomLeftRadius": undefined,
          "borderBottomRightRadius": undefined,
          "borderBottomStartRadius": undefined,
          "borderBottomWidth": undefined,
          "borderColor": undefined,
          "borderEndColor": undefined,
          "borderEndWidth": undefined,
          "borderLeftColor": undefined,
          "borderLeftWidth": undefined,
          "borderRadius": undefined,
          "borderRightColor": undefined,
          "borderRightWidth": undefined,
          "borderStartColor": undefined,
          "borderStartWidth": undefined,
          "borderStyle": undefined,
          "borderTopColor": undefined,
          "borderTopEndRadius": undefined,
          "borderTopLeftRadius": undefined,
          "borderTopRightRadius": undefined,
          "borderTopStartRadius": undefined,
          "borderTopWidth": undefined,
          "borderWidth": undefined,
          "bottom": undefined,
          "end": undefined,
          "height": undefined,
          "left": undefined,
          "margin": 10,
          "marginBottom": undefined,
          "marginEnd": undefined,
          "marginHorizontal": undefined,
          "marginLeft": undefined,
          "marginRight": undefined,
          "marginStart": undefined,
          "marginTop": undefined,
          "marginVertical": undefined,
          "maxHeight": undefined,
          "maxWidth": undefined,
          "minHeight": undefined,
          "minWidth": undefined,
          "opacity": undefined,
          "overflow": undefined,
          "padding": undefined,
          "paddingBottom": undefined,
          "paddingEnd": undefined,
          "paddingHorizontal": undefined,
          "paddingLeft": undefined,
          "paddingRight": undefined,
          "paddingStart": undefined,
          "paddingTop": undefined,
          "paddingVertical": undefined,
          "position": undefined,
          "right": undefined,
          "start": undefined,
          "top": undefined,
          "width": undefined,
          "zIndex": undefined,
        },
        undefined,
      ]
    }
  >
    <Text
      style={
        [
          {
            "fontSize": 20,
            "fontWeight": "500",
            "letterSpacing": 0.15,
          },
          {
            "color": "#000000",
          },
          undefined,
        ]
      }
    >
      Loading
    </Text>
  </View>
</View>
`;
